type_inference/type_retrieval_service.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/python
+#
+# Copyright 2023 Logica Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import cache
+from typing import Dict
+from type_inference.postgresql_type_parser import try_parse_postgresql_type
+import psycopg2
+from os import linesep
+
+from type_inference.type_retrieval_exception import TypeRetrievalException
+
+built_in_types = set()
```

Comment:
Let's avoid global variables. Please create a BuiltInTypeRetriever class, and make this a class-level variable. The class would include InitBuiltInTypes and UnpackType class-methods.